### PR TITLE
Refactor/file io tests

### PR DESCRIPTION
**Pull request by jphill19**
Test were added and created for StatTracker, game, season, team.

Also refactored code inside StatTracker to solve the bug that arose from tests.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -33,10 +31,12 @@ def self.teams_reader(csv_data)
     
     def self.game_teams_reader(csv_data)
         seasons = Hash.new(0)
+        count = 1 
         CSV.foreach(csv_data, headers: true, header_converters: :symbol) do |row|
-            seasons[row[:game_id].to_i] = Season.new(row)
+            seasons[count] = Season.new(row)
```
Comment (jchirch) [inline lib/stat_tracker.rb:36]: Great way to bypass this bug and access all rows even if they share a game_id.

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+
```
Comment (jchirch) [inline spec/game_spec.rb:2]: Solid coverage on initializing objects and all of their attributes. Can't think of any ways to be more thorough at this time.

File: spec/stat_tracker_spec.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            seasons_data = StatTracker.game_teams_reader(@locations[:game_teams])
+
+            expect(seasons_data.class).to eq Hash
+            expect(seasons_data[1]).to be_an_instance_of Season
+        end
+
+        it "each season object it creates, all their attributes are truthy" do
+            seasons_data = StatTracker.game_teams_reader(@locations[:game_teams])
+            
+            seasons_data.each do |game_id, season_object|
```
Comment (jchirch) [inline spec/stat_tracker_spec.rb:80]: Brilliant way to test every row in 1 test. I think utilizing an enumerable in our tests are a fantastic way to guarantee coverage and will benefit everyone in the group's learning.

**Review by jchirch — COMMENTED** [review]
Overall the code is brilliant and very organized. It may be personal preference, but I would suggest making more commits for this amount of new code.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (bwillett2003) [reply lib/stat_tracker.rb:36]: I second this. It found a work around to the data coming to us.

**Pull request merged**